src/Http/Handler/ResponseHandlerSignature.php

```diff
@@ -168,7 +168,7 @@ private function ensureHeaderIsCorrectlyCased(string $headerName)
         if ($regexResult != self::REGEX_CHECK_FAILED) {
             foreach ($matches[self::INDEX_FIRST] as $match) {
                 $matchUpper = strtoupper($match);
-                $headerName = preg_replace(vsprintf(self::REGEX_REPLACE, $match), $matchUpper, $headerName);
+                $headerName = preg_replace(vsprintf(self::REGEX_REPLACE, [$match]), $matchUpper, $headerName);
```

Comment:
This does the same
```suggestion
                $headerName = preg_replace(sprintf(self::REGEX_REPLACE, $match), $matchUpper, $headerName);
```

Reply:
@MathijsvanVelde yes, do you want me to update the PR with that suggestion?